frontend/projects/upgrade/src/app/core/feature-flags/store/feature-flags.selectors.ts

```diff
@@ -53,7 +53,23 @@ export const selectSelectedFeatureFlag = createSelector(
     // be very defensive here to make sure routerState is correct
     const flagId = routerState?.state?.params?.flagId;
     if (flagId) {
-      return featureFlagState.entities[flagId];
+      const featureFlag = featureFlagState.entities[flagId];
```

Comment:
This selector should be kept as simple as possible, it's a very widely used and just needs to access the flag, we already have selectors for the lists that derive from this base selector.